src/integration-tests/src/bigquery.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Result;
+use gax::error::Error;
+use rand::{Rng, distr::Alphanumeric};
+
+pub async fn dataset_admin(
+    builder: bigquery::builder::dataset_service::ClientBuilder,
+) -> Result<()> {
+    // Enable a basic subscriber. Useful to troubleshoot problems and visually
+    // verify tracing is doing something.
+    #[cfg(feature = "log-integration-tests")]
+    let _guard = {
+        use tracing_subscriber::fmt::format::FmtSpan;
+        let subscriber = tracing_subscriber::fmt()
+            .with_level(true)
+            .with_thread_ids(true)
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
+            .finish();
+
+        tracing::subscriber::set_default(subscriber)
+    };
+
+    let project_id = crate::project_id()?;
+    let client = builder.build().await?;
+    cleanup_stale_datasets(&client, &project_id).await?;
+
+    let rand_suffix: String = rand::rng()
+        .sample_iter(&Alphanumeric)
+        .take(8)
+        .map(char::from)
+        .collect();
+
+    let ds_name = format!("rust_bq_test_dataset_{rand_suffix}");
+
+    println!("CREATING DATASET WITH ID: {ds_name}");
+
+    let create = client
+        .insert_dataset(&project_id)
+        .set_dataset(bigquery::model::Dataset::new().set_dataset_reference(
+            bigquery::model::DatasetReference::new().set_dataset_id(&ds_name),
+        ))
+        .send()
+        .await?;
+    println!("CREATE DATASET = {create:?}");
+
+    assert!(create.dataset_reference.is_some());
+
+    let list = client.list_datasets(&project_id).send().await?;
+    println!("LIST DATASET = {} entries", list.datasets.len());
+
+    assert!(!list.datasets.is_empty());
+    assert!(list.datasets.len() > 1);
+    assert!(
+        list.datasets
+            .iter()
+            .find(|v| v.id.contains(&ds_name))
+            .is_some()
+    );
```

Comment:
All of this can be changed to:

```suggestion
    assert!(
        list.datasets
            .iter()
            .find(|v| v.id.contains(&ds_name))
            .is_some(),
            "{:?}", list.datasets
    );
```

Separately, why "contains"? Can we make that a more specific predicate?

Reply:
the Dataset `id` contains the fully qualified id of the dataset, so it would be something like `projects/{projectId}/datasets/{datasetId}`, so that's why I used `contains`.

Reply:
Yes, I was suggesting we do something more specific, such as:

```rust
  v.id.strip_suffix(&ds_name).and(Some(true)).unwrap_or_default()
```

or
```rust
  v.id == format!("projects/{project_id}/datasets/{ds_name}")
```

or (gross):

```rust
  v.id.strip_prefix("projects/").and_then(|s| s.strip_prefix(&project_id)).and_then(|s|  s.strip_prefix("/datasets/")).and_then(|s| s == &ds_name).unwrap_or_default()
```